handlers.go

```diff
@@ -223,12 +206,12 @@ func mkVmap(keys []string, values []interface{}) (*map[string]interface{}, error
 		// convert from sql.RawBytes to string
 		rbp, ok := values[i].(*sql.RawBytes)
 		if !ok {
-			return &ret, fmt.Errorf("sql conversion error")
+			return ret, fmt.Errorf("sql conversion error")
 		}
 		s := string(*rbp)
 		ret[keys[i]] = interface{}(&s)
 	}
-	return &ret, nil
+	return ret, nil
 }
 
 // mkSQLRow() returns a list of interface{} of the given length,
```

Comment:
thanks for this insight.  you're right that my code has too much interface conversion going on.  in this case, though, the problem is that the vals slice returned from mkSQLRow() will be passed to rows.Scan(vals...) which is expecting its arguments to be of type interface{}.
https://golang.org/pkg/database/sql/#Row.Scan

any suggestion on how to finesse this issue?